ang/civiawards/shared/configs/case-type-buttons.config.js

```diff
@@ -1,29 +1,35 @@
-(function (angular, _, url) {
+(function (angular, _, checkPermission, url) {
   var module = angular.module('civiawards');
   var AWARDS_CATEGORY_NAME = 'awards';
   var AWARD_CONFIG_URL = 'civicrm/a/#/awards/';
+  var EDIT_AWARD_PERMISSION = 'create/edit awards';
 
   module.config(function (CaseTypeProvider, CaseTypeCategoryProvider, DashboardCaseTypeButtonsProvider) {
     var awardCategory = CaseTypeCategoryProvider.findByName(AWARDS_CATEGORY_NAME);
     var awardCaseTypes = CaseTypeProvider.getByCategory(awardCategory.value);
+    var canEditAwards = checkPermission(EDIT_AWARD_PERMISSION);
 
-    addConfigurationButtonsToCaseTypes(awardCaseTypes);
+    (function () {
+      if (!canEditAwards) {
+        return;
+      }
+
+      _.forEach(awardCaseTypes, addConfigurationButtonsToCaseType);
```

Comment:
Why did we took the loop out of the function?

Reply:
makes the function itself simpler. It also explains what happens to each award.

Imagine we need to filter by awards from last year first before adding the configuration button (a crazy example, bear with me). We can have  this on the `init` function:

```js
_.chain(awardCaseTypes)
  .filter(getLastYearAwards)
  .forEach(addConfigurationButtonsToCaseType);
```

This explains very well how we assign the configuration buttons.

It also allows us to use the function for a single object: `addConfigurationButtonsToCaseType(myAward)` or for multiple ones: `myAwards.forEach(addConfigurationButtonsToCaseType)`. With the previous implementation it was all or nothing.